### PR TITLE
Add OrderSchema tests

### PR DESCRIPTION
**Pull request by fmrsabino**
## Summary

- Adds unit tests to `OrderSchema`.

## Changes

- Changes `.default('unknown')` to `.catch('unknown')` in the `OrderSchema` as the default value would only be set if no value was provided. If an unknown value is provided we should still set the respective value as `unknown`.

### CODE REVIEW

File: src/domain/swaps/entities/schemas/order.schema.ts [hunk 1]
```diff
@@ -62,7 +62,7 @@ export const OrderSchema = z.object({
           'PreValidationError',
           'unknown',
         ])
-        .default('unknown')
+        .catch('unknown')
         .nullish()
         .default(null),
```
Comment (iamacook) [inline src/domain/swaps/entities/schemas/order.schema.ts:67]:
```suggestion
        .catch('unknown'),
```

**Review by iamacook — CHANGES_REQUESTED** [review]
Can we also add test conditions for:

- Address checksumming of `sellToken`, `buyToken`, `receiver`, `from`, `owner`, `onChainUser` and `onchainUser.sender`
- Defaulting to `null` for `receiver`, `from`, `quoteId`, `availableBalance`, `ethflowData`, `onchainUser`, `executedSurplusFee` and `fullAppData`.

I've also noticed that there are some hexadecimal values that could use the `HexSchema` in the `OrderSchema` that could then be tested as well: `signature` and `ethflowData.refundTxHash`.

File: src/domain/swaps/entities/schemas/order.schema.spec.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+    const result = OrderSchema.safeParse(order);
+
+    expect(result.success).toBe(true);
+  });
+
+  it('hexadecimal signature should be valid', () => {
+    const order = {
+      ...orderBuilder().build(),
+      signature: faker.string.hexadecimal() as `0x${string}`,
```
Comment (iamacook) [inline src/domain/swaps/entities/schemas/order.schema.spec.ts:19]: Is there a reason to now use `with` in these tests?

File: src/domain/swaps/entities/schemas/order.schema.spec.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+  it('should fallback to unknown kind on order with an invalid kind', () => {
+    const order = { ...orderBuilder().build(), kind: faker.string.sample() };
+
+    const result = OrderSchema.safeParse(order);
+
+    expect(result.success && result.data.kind).toBe('unknown');
+  });
+
+  it.each<keyof Order>([
```
Comment (iamacook) [inline src/domain/swaps/entities/schemas/order.schema.spec.ts:114]: Nice type usage!

**Review by iamacook — APPROVED** [review]

File: src/domain/swaps/entities/schemas/order.schema.spec.ts (hunk 2 above)
Comment (fmrsabino) [reply src/domain/swaps/entities/schemas/order.schema.spec.ts:19]: No reason! Will fix it! 👍

Comment (fmrsabino) [reply src/domain/swaps/entities/schemas/order.schema.spec.ts]: I've fixed it in [107cf08](https://github.com/safe-global/safe-client-gateway/pull/1302/commits/107cf0878fbb2dead19a4885feadb58ec472fd1c).

Additionally I merged the "unknown" tests into one (parameterised test).

**Review by iamacook — APPROVED** [review]

**Pull request merged**